src/components/dialog/dialog.pw.tsx

```diff
@@ -0,0 +1,596 @@
+    mount,
+    page,
+  }) => {
+    await mount(
+      <DialogComponent title="Sample Dialog" help="Some help text" />
+    );
+
+    const helpIcon = page.getByLabel("help");
+    await helpIcon.hover();
+
+    await expect(page.getByRole("tooltip")).toHaveText("Some help text");
+  });
+
+  test("when Dialog is opened, the first focusable element should be focused", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogWithFirstFocusableElement />);
+    await expect(
+      page.getByRole("button").filter({ hasText: "Press me" })
+    ).toBeFocused();
+  });
+
+  test("when disableAutoFocus prop is passed, the first focusable element should not be focused", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogWithFirstFocusableElement disableAutoFocus />);
+    await expect(
+      page.getByRole("button").filter({ hasText: "Press me" })
+    ).not.toBeFocused();
+  });
+
+  test("when tabbing through Dialog content, focus should remain trapped inside the Dialog", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent />);
+    const dialog = page.getByRole("dialog");
+    const thirdTextbox = page.getByLabel("Textbox3");
+    const secondTextbox = page.getByLabel("Textbox2");
+    const firstTextbox = page.getByLabel("Textbox1");
+    const closeButton = page.getByLabel("Close");
+
+    await dialog.press("Tab");
+    await expect(closeButton).toBeFocused();
+
+    await closeButton.press("Tab");
+    await expect(firstTextbox).toBeFocused();
+
+    await firstTextbox.press("Tab");
+    await expect(secondTextbox).toBeFocused();
+
+    await secondTextbox.press("Tab");
+    await expect(thirdTextbox).toBeFocused();
+
+    await thirdTextbox.press("Tab");
+    await expect(closeButton).toBeFocused();
+  });
+
+  test("when shift tabbing through Dialog content, focus should remain trapped inside the Dialog", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent />);
+    const dialog = page.getByRole("dialog");
+    const thirdTextbox = page.getByLabel("Textbox3");
+    const secondTextbox = page.getByLabel("Textbox2");
+    const firstTextbox = page.getByLabel("Textbox1");
+    const closeButton = page.getByLabel("Close");
+
+    await dialog.press("Shift+Tab");
+    await expect(thirdTextbox).toBeFocused();
+
+    await thirdTextbox.press("Shift+Tab");
+    await expect(secondTextbox).toBeFocused();
+
+    await secondTextbox.press("Shift+Tab");
+    await expect(firstTextbox).toBeFocused();
+
+    await firstTextbox.press("Shift+Tab");
+    await expect(closeButton).toBeFocused();
+
+    await closeButton.press("Shift+Tab");
+    await expect(thirdTextbox).toBeFocused();
+  });
+
+  test("when tabbing through Dialog content, background should not scroll to the bottom of the page", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogBackgroundScrollTest />);
```

Comment:
suggestion: (non-blocking) it would be better to add a space to separate the mount from the assertion to follow the same pattern as with the other test cases.